### PR TITLE
rst_formatted help output

### PR DESCRIPTION
**Pull request by ceriottm**
This is a preliminary version - it needs to be tweaked also depending on how it will be integrated into the general docs. 
The PR contains a docs/test_rst.py file that generates a test example, but the file should be removed before merging

### CODE REVIEW

**Comment by ceriottm** [comment]
Not perfect, but IMO good enough. Now all the helpstrings for input fields and property outputs are pulled into the online docs. Also rationalized a bit the latex side of things, and cleared quite a few compilation errors.

File: doc/src/conf.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 author = "The i-PI developers"
 
 # The full version, including alpha/beta/rc tags
-release = "TODO VERSION"
+release = "2"
```
[the remark this answers is not in the document]
Comment (ceriottm) [reply doc/src/conf.py]: Fair. Done. sphynx is much easier to use than I expected.

**Comment by venkatkapil24** [comment]
Looks like make-ing the docs requires sphinx-build which is not in the requirements.txt file. @ceriottm which version are you using or can you add it to the requirements.txt file?

**Comment by venkatkapil24** [comment]
> Looks like make-ing the docs requires sphinx-build which is not in the requirements.txt file. @ceriottm which version are you using or can you add it to the requirements.txt file?

okay, my bad, I can see there is a separate requirements.txt file.

**Comment by venkatkapil24** [comment]
Overall, the html version of docs look very nice and user-friendly. Well done @ceriottm and @Luthaf :) Nonetheless, here are some comments

1. We need to add a README or to add something to the main README here to ensure people install the correct versions of sphinx etc. 

2. It would be nice to also tell them that for html they can do make html and open the index.html file in _build/html/

3. And for the pdf version they can do make latex and open the I-pi.pdf file in _build/latex/

4. Need to add a version number 
image

5. The i-PI logo seems to be missing the letters and it doesn't appear in the i-pi.pdf manual. 

image

**Comment by venkatkapil24** [comment]
6. The features on the website and the manual are not consistent. E.g. the committee model is missing in the docs. 

7. Could vibrational self-consistent field and self-consistent phonons (https://pubs.acs.org/doi/pdf/10.1021/acs.jctc.9b00596) be added to the features website? @mahrossi

**Comment by ceriottm** [comment]
> 6. The features on the website and the manual are not consistent. E.g. the committee model is missing in the docs.

What do you mean? FFCommittee is there. You mean a more detailed discussion of the committees?

> 7. Could vibrational self-consistent field and self-consistent phonons (https://pubs.acs.org/doi/pdf/10.1021/acs.jctc.9b00596) be added to the features website? @mahrossi

You can add them yourself ^_^

**Comment by ceriottm** [comment]
> Overall, the html version of docs look very nice and user-friendly. Well done @ceriottm and @Luthaf :) Nonetheless, here are some comments
> 
> 1. We need to add a README or to add something to the main README here to ensure people install the correct versions of sphinx etc.
Added 

> 2. It would be nice to also tell them that for html they can do make html and open the index.html file in _build/html/
Sure

> 3. And for the pdf version they can do make latex and open the I-pi.pdf file in _build/latex/
Make latex didn't work so I just pass it on to the old-style tex

> 4. Need to add a version number
It actually should be already there 
image

> 5. The i-PI logo seems to be missing the letters and it doesn't appear in the i-pi.pdf manual.
> 
> image

I think the PDF version is something we should tackle in a separate PR. Maintaining two sets of files is a bad idea, so we should deprecate the old style latex and make sure that the tex generated from the RST works at least as well.

**Comment by mahrossi** [comment]
I think this can/should be merged. I am adding an approving review. I am puzzled as to why Venkat was still seeing the "TODO" bug, that had already been removed! Anyhow, I don't see it. Looks nice. Re. features @venkatkapil24 I try to go back to the website soon - I finished my edits there some days ago. If you do it, let me know.

**Review by mahrossi — APPROVED** [review]
Looks like a great great improvement and details have been tidied up.

**Pull request merged**